### PR TITLE
Add Bluetooth Interface

### PR DESCRIPTION
**Pull request by amc1234**
**Purpose**
Add bluetooth declarations and macros so the Raspberry Pi can interface with Android.

**Summary**
Python code was developed with pre-declared functions that can be used as necessary for Wally. This makes connecting/disconnecting and sending data easier.

This works differently than our original design since this is made for bluetooth classic (the Raspberry Pi 3 uses bluetooth classic), but has the option to switch to BLE though that has yet to be tested.

**Testing Done**
Tested on Android devices: Google Pixel 3A

Bluetooth terminal app from the Play Store (Serial Bluetooth Terminal): https://github.com/kai-morich/SimpleBluetoothTerminal/tree/49de1130a93769796e00938aed30ddb49785dccf

- Checked that the Raspberry Pi can connect/disconnect to smartphone
- Checked that the smartphone can send data to the Raspberry Pi
- Checked that the Raspberry Pi can send data to the smartphone

**Testing To Do**
- Test that the functions work well with our main code
- Might be able to do a scan instead of a pre-pairing through bluetoothctl

### CODE REVIEW

**Review by tiger-hao — COMMENTED** [review]
Not a fan of the formatting. Can we standardize the format?

If you're using vscode, I'd use [these settings](https://stackoverflow.com/questions/45823734/visual-studio-code-formatting-for).

**Comment by rs2thoma** [comment]
> Not a fan of the formatting. Can we standardize the format?
> 
> If you're using vscode, I'd use [these settings](https://stackoverflow.com/questions/45823734/visual-studio-code-formatting-for).

yeah the indentation before the brace is weird. But newline for brace >>> brace on same line

**Comment by rs2thoma** [comment]
can we rename the .c file to a .cpp file since we're using c++ and compile with g++? it should still run the same. If it  doesn't you could always surround the whole file with something like
`
#ifdef __cplusplus
extern "C"
{
#endif

// ...

#ifdef __cplusplus
}
#endif
`

**Comment by rs2thoma** [comment]
I think .c(pp) file should be under src/ and .h should be under inc/

**Comment by rs2thoma** [comment]
Did you change the .c or .h file from what the source was? If so you should note it somewhere

**Review by rs2thoma — APPROVED** [review]
python is EZ lol
also we should standardize things, i.e. tabs vs spaces, and the python version

File: BLE.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#If not previously paired with an Android device it must be using bluetooth packages
+#sudo bluetoothctl
+#power on
+#discoverable on
+#pairable on
+#pair <MAC_ADDRESS> - pi must respond yes
+
+import bluetooth
+
+class socket:
```
Comment (tiger-hao) [inline BLE.py]: class names should be capitalized

File: BLE.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        socket.server = bluetooth.BluetoothSocket( bluetooth.RFCOMM )
+        port = 1
+        socket.server.bind(("",port))
+
+    def advertise(socket):
+        socket.server.listen(1)
+
+    def connect(socket):
+        socket.client, address = socket.server.accept()
+        print ("Accepted connection from ",address)
```
Comment (tiger-hao) [inline BLE.py]: `print` automatically adds a space in between items, so you don't need the space after "from"

File: README.md [hunk 3]
```diff
@@ -1 +1,11 @@
-# Firmware
\ No newline at end of file
+# Firmware
+
+**Bluetooth**
+The following packages must be installed:
+```sudo apt-get install bluetooth libbluetooth-dev bluez```
+```sudo pip3 install pybluez```(might not be necessary)
```
Comment (tiger-hao) [inline README.md]: just fyi ``` is multiline so you can do
```
sudo apt
sudo apt
```
to make it one big block

Comment (tiger-hao) [reply README.md]: er, it formats comments hehe
'''
sudo
sudo
'''

**Comment by tiger-hao** [comment]
> also we should standardize things, i.e. tabs vs spaces, and the python version

- [Developers Who Use Spaces Make More Money Than Those Who Use Tabs](https://stackoverflow.blog/2017/06/15/developers-use-spaces-make-money-use-tabs/)

- python3.8+ should be good

**Review by tiger-hao — APPROVED** [review]

**Pull request merged**